subprojects/core/src/main/java/org/gradle/api/internal/plugins/ExtensionsStorage.java

```diff
@@ -65,7 +67,10 @@ public ExtensionsSchema getSchema() {
 
     public <T> T configureExtension(String name, Action<? super T> action) {
         ExtensionHolder<T> extensionHolder = uncheckedCast(extensions.get(name));
-        return extensionHolder.configure(action);
+        if (extensionHolder != null) {
+            return extensionHolder.configure(action);
+        }
+        throw new UnknownDomainObjectException(extensionWithNameDoesNotExistMessage(name));
```

Comment:
:x: Extract the whole exception instantiation expression since it's the same on both places.

`throw unknownExtensionException(name);`

Reply:
I'm happy to do so, but doesn't that obscure the thrown site by one level in the stack trace?

Reply:
Yes. The shorter and cleaner version of the code is worth it.